### PR TITLE
Adding getters for private variables in HandFinder

### PR DESCRIPTION
**Pull request by maxzieba**
## Proposed changes

User Story: https://shadowrobot.atlassian.net/browse/SP-389
Adding getters for private variables in HandFinder as part of lint fixes.

## Checklist

Before posting a PR ensure that from each of the below categories **AT LEAST ONE BOX HAS BEEN CHECKED**. If more than one category is applicable then more can be checked. Also ensure that the proposed changes have been filled out with relevant information for reviewers.

## Tests

- [X] No tests required to be added. (For small changes that will be tested by CI/CD infrastructure).
- [ ] Added/Modified automated and PhantomHand CI tests (if a new class is added (Python or C++), the interface of that class must be unit tested).
- [ ] Manually tested in simulation (if simulation specific or no hardware required to test the functionality). 
- [ ] Manually tested on hardware (if hardware specific or related).

## Documentation

- [X] No documentation required to be added.
- [ ] Added documentation (For any new feature, explain what it does and how to use it. Write the documentation in a relevant space, e.g. Github, Confluence, etc).
- [ ] Updated documentation (For changes to pre-existing features mentioned in the documentation).

### CODE REVIEW

File: sr_utilities/scripts/sr_utilities/hand_finder.py [hunk 1]
```diff
@@ -141,6 +141,18 @@ def wait_for_hand_params(self, timeout_in_secs):
             self._hand_h = True
             self._hand_h_parameters = rospy.get_param("/fh_hand")
 
+    def is_type_hand_e(self):
+        return self._hand_e
+
+    def get_hand_e_parameters(self):
+        return self._hand_parameters
```
Comment (jeferal) [inline sr_utilities/scripts/sr_utilities/hand_finder.py]: Is this variable only related to hand e?

Comment (maxzieba) [reply sr_utilities/scripts/sr_utilities/hand_finder.py]: The class had only hand_h and hand_e related params, so I didnt extend with lite etc.
hand_parameters is for hand_e, hand_e_parameters is for hand_h (you can check in init of the class)

File: sr_utilities/scripts/sr_utilities/hand_finder.py [hunk 2]
```diff
@@ -141,6 +141,18 @@ def wait_for_hand_params(self, timeout_in_secs):
             self._hand_h = True
             self._hand_h_parameters = rospy.get_param("/fh_hand")
 
+    def is_type_hand_e(self):
+        return self._hand_e
```
Comment (bdenoun) [inline sr_utilities/scripts/sr_utilities/hand_finder.py]: please do change these variables as they don't come out as booleans

File: sr_utilities/scripts/sr_utilities/hand_finder.py (hunk 1 above)
Comment (bdenoun) [reply sr_utilities/scripts/sr_utilities/hand_finder.py]: If so, please make this variable more meaningful

File: sr_utilities/scripts/sr_utilities/hand_finder.py [hunk 3]
```diff
@@ -141,6 +141,18 @@ def wait_for_hand_params(self, timeout_in_secs):
             self._hand_h = True
             self._hand_h_parameters = rospy.get_param("/fh_hand")
 
+    def is_type_hand_e(self):
```
Comment (bdenoun) [inline sr_utilities/scripts/sr_utilities/hand_finder.py]: please do update the date of the licence of this file

File: sr_utilities/scripts/sr_utilities/hand_finder.py [hunk 4]
```diff
@@ -141,6 +141,18 @@ def wait_for_hand_params(self, timeout_in_secs):
...
+        return self._hand_e
+
+    def get_hand_e_parameters(self):
```
Comment (bdenoun) [inline sr_utilities/scripts/sr_utilities/hand_finder.py]: Why not using the property decorator to actual make getters and setters?

**Review by bdenoun — CHANGES_REQUESTED** [review]

File: sr_utilities/scripts/sr_utilities/hand_finder.py (hunk 4 above)
Comment (maxzieba) [reply sr_utilities/scripts/sr_utilities/hand_finder.py]: as this is not yet our standard so this has a better readability when reused elsewhere in my opinion. If this is not ok for you, are you happy with me changing this to:

@property
def hand_e_parameters(self):
        return self._hand_parameters
        
@property
def hand_h_parameters(self):
        return self._hand_h_parameters
        
?

File: sr_utilities/scripts/sr_utilities/hand_finder.py (hunk 1 above)
Comment (bdenoun) [reply sr_utilities/scripts/sr_utilities/hand_finder.py]: > The class had only hand_h and hand_e related params, so I didnt extend with lite etc. hand_parameters is for hand_e, hand_h_parameters is for hand_h (you can check in init of the class)

This does not make sense does it?

File: sr_utilities/scripts/sr_utilities/hand_finder.py (hunk 4 above)
Comment (bdenoun) [reply sr_utilities/scripts/sr_utilities/hand_finder.py]: I don't think it's a matter of standard ^^ If we want to add a getter or setter, let's simply do it right :). In addition, you should also have a setter I guess

File: sr_utilities/scripts/sr_utilities/hand_finder.py (hunk 1 above)
Comment (maxzieba) [reply sr_utilities/scripts/sr_utilities/hand_finder.py]: sorry, edited (hand_e_parameters -> hand_h_parameters)

Comment (rnzenha-s) [reply sr_utilities/scripts/sr_utilities/hand_finder.py]: It seems the issue is that code elsewhere is still trying to access this parameter directly. Once all of it is changed to use the new getter this parameter could be changed safely

File: sr_utilities/scripts/sr_utilities/hand_finder.py (hunk 4 above)
Comment (maxzieba) [reply sr_utilities/scripts/sr_utilities/hand_finder.py]: I'd say its a matter of standard to keep it consistent across all codes (either decorators or methods set/get, or even something else) or no? I'll add only getters as, this is supposed to be initialized from the parameter server.

File: sr_utilities/scripts/sr_utilities/hand_finder.py [hunk 5]
```diff
@@ -141,6 +141,9 @@ def wait_for_hand_params(self, timeout_in_secs):
             self._hand_h = True
             self._hand_h_parameters = rospy.get_param("/fh_hand")
 
+    def get_hand_h_parameters(self):
+        return self._hand_h_parameters
```
Comment (rnzenha-s) [inline sr_utilities/scripts/sr_utilities/hand_finder.py:147]: I would add:
```
  if not self._hand_h:
      rospy.logerr("No Hand H present - can't get hand parameters")
```
to keep consistency.
Also remove 'object' from HandConfig(object) pls!

**Review by rnzenha-s — CHANGES_REQUESTED** [review]

File: sr_utilities/scripts/sr_utilities/hand_finder.py (hunk 5 above)
Comment (maxzieba) [reply sr_utilities/scripts/sr_utilities/hand_finder.py:147]: Good point, added logerr and removed objects :)

File: sr_utilities/scripts/sr_utilities/hand_finder.py (hunk 4 above)
Comment (bdenoun) [reply sr_utilities/scripts/sr_utilities/hand_finder.py]: Even if it's set from the parameter server, you will do `self.param = blabla`, which would call the setter  ^^. And the difference between decorator and simple getter and setter runs much deeper than "standard"

**Review by rnzenha-s — APPROVED** [review]

**Review by jeferal — APPROVED** [review]

File: sr_utilities/scripts/sr_utilities/hand_finder.py (hunk 4 above)
Comment (toliver) [reply sr_utilities/scripts/sr_utilities/hand_finder.py]: Let's add this to the discussion issue list in the next SW team meeting, please.

Comment (bdenoun) [reply sr_utilities/scripts/sr_utilities/hand_finder.py]: So we merge?

**Comment by bdenoun** [comment]
I created a story to address my comment in the future: https://shadowrobot.atlassian.net/browse/SP-450

**Review by bdenoun — APPROVED** [review]
